src/QSymbolicsBase/predefined.jl

```diff
@@ -205,23 +205,28 @@ function Base.show(io::IO, x::SProjector)
     print(io,"]")
 end
 
-"""Dagger a Ket into Bra."""
-@withmetadata struct SDagger <: Symbolic{AbstractBra}
-    ket::Symbolic{AbstractKet}
+"""Dagger, i.e., adjoint of quantum objects (kets, bras, operators, and inner products)"""
+@withmetadata struct SDagger{T<:Union{Complex, QObj}} <: Symbolic{Union{Complex, T}}
```

Comment:
I do not think we should permit `Complex` as a type parameter here. Any reason for adding it?

Reply:
My mistake, just fixed it.